node-notes-cli/app.js

```diff
@@ -0,0 +1,87 @@
+import { readFile, writeFile } from 'node:fs/promises';
+
+const argVar = process.argv[2];
+const content = process.argv[3];
+const content2 = process.argv[4];
+
+if (argVar === 'read' && content === undefined) {
+  try {
+    const response = await readFile('./data.json', 'utf8');
+    const data = JSON.parse(response);
+    if (data && data.notes) {
+      for (const key in data.notes) {
+        if (Object.getOwnPropertyDescriptor(data.notes, key)) {
+          const noteValue = data.notes[key];
+          console.log(`${key}: ${noteValue}`);
+        }
+      }
+    } else {
+      console.error(
+        'Invalid JSON format or missing "notes" object in the file.'
+      );
+    }
+  } catch (err) {
+    console.error('Error reading or parsing the file:', err);
+  }
+}
+
+if (argVar === 'create' && content !== undefined) {
+  try {
+    // Read the existing data from the file
+    const response = await readFile('./data.json', 'utf8');
```

Comment:
I like that you read at the top of each of these cases. That's similar to how it will be done when we introduce a database. But, it would also be nice to not duplicate this line of code (and the next one) over and over. (In other words, create a function you can call.)